### PR TITLE
e2e + #195 test

### PR DESCRIPTION
**Pull request by escritorio-gustavo**
@NyxCode I have copied over the code from #195 to check it if'd be able to pass the e2e tests. With a change to the `dependencies` test (adding `cargo t -p dependency1`) it works!

### CODE REVIEW

**Comment by escritorio-gustavo** [comment]
Note: **all** changes outside `e2e` are from #195

File: e2e/dependencies/dependency1/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,12 @@
 use ts_rs::TS;
 
 #[derive(TS)]
+#[ts(export)]
```
Comment (escritorio-gustavo) [inline e2e/dependencies/dependency1/src/lib.rs:4]: Lib authors must use `#[ts(export)]`

File: e2e/dependencies/dependency1/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
 pub struct LibraryType {
     pub a: i32
-}
\ No newline at end of file
+}
+
+#[test]
+fn env_set() {
+    assert_ne!(env!("TS_RS_EXPORT_DIR"), "");
+}
```
Comment (escritorio-gustavo) [inline e2e/dependencies/dependency1/src/lib.rs:12]: It is possible to enforce that the consumer sets the environment variable

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -12,6 +12,7 @@ jobs:
       - name: dependencies e2e test
         working-directory: e2e/dependencies/consumer
         run: |
+          cargo t -p dependency1
```
Comment (escritorio-gustavo) [inline .github/workflows/test.yml:15]: Need to run the dependencies' tests, luckily this only needs to be done on dependency install/update

File: e2e/workspace/.cargo/config.toml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[env]
+TS_RS_EXPORT_DIR = { value = "./parent/bindings/", relative = true }
```
Comment (escritorio-gustavo) [inline e2e/workspace/.cargo/config.toml:2]: The path is relative to `.cargo`'s parent

**Review by escritorio-gustavo — COMMENTED** [review]
How it works:

**Comment by escritorio-gustavo** [comment]
Btw, this makes #211 obsolete, so if this gets merged that PR should be closed

**Comment by NyxCode** [comment]
Awesome!
(Please correct me if I get any of this wrong)
To me, it seems like these are the tradeoffs for this solution:
- **CON** library types must be `#[ts(export)]`.
- **CON** All library types will be exported, not just those used in the consumer
- **PRO** No locking is required, since every type is only ever exported by one test
- **---** Setting the env variables and running `cargo t -p` for every dependency is a bit cumbersome, so this solution would profit from a CLI tool that automatically does all that

**Comment by escritorio-gustavo** [comment]
Yes, those are all correct

**Comment by escritorio-gustavo** [comment]
> - **CON** All library types will be exported, not just those used in the consumer

If we create a CLI to run the tests, maybe it could filter only the types used in the consumer?

**Pull request closed without merging**